File: backend/models/family.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from config.database import Base


class Family(Base):
    __tablename__ = "families"

    id = Column(Integer, primary_key=True, index=True)
    family_name = Column(String, nullable=False)
    description = Column(String, nullable=True)
    created_by = Column(Integer, ForeignKey("users.id"))
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    # 关系
    members = relationship("FamilyMember", back_populates="family", cascade="all, delete-orphan")
    creator = relationship("User", foreign_keys=[created_by])
    categories = relationship("BillCategory", back_populates="family")
    bills = relationship("Bill", back_populates="family")


class FamilyMember(Base):
    __tablename__ = "family_members"

    id = Column(Integer, primary_key=True, index=True)
    family_id = Column(Integer, ForeignKey("families.id"))
    user_id = Column(Integer, ForeignKey("users.id"))
    role = Column(String, default="member")  # owner, admin, member
    joined_at = Column(DateTime(timezone=True), server_default=func.now())

    # 关系
    family = relationship("Family", back_populates="members")
    user = relationship("User")

File: frontend/src/types/family.ts
export interface Family {
  id: number;
  family_name: string;
  description?: string;
  created_by: number;
  created_at: string;
  updated_at: string;
}

export interface FamilyMember {
  id: number;
  family_id: number;
  user_id: number;
  role: 'admin' | 'member';
  created_at: string;
  user: {
    id: number;
    username: string;
    full_name?: string;
    email: string;
  };
}

export interface FamilyCreate {
  family_name: string;
  description?: string;
  invite_usernames?: string[];
}

export interface FamilyUpdate {
  family_name?: string;
  description?: string;
}

export interface UserSearchResult {
  id: number;
  username: string;
  full_name?: string;
  email: string;
}

export interface FamilyListResponse {
  families: Family[];
}

export interface FamilyMemberListResponse {
  members: FamilyMember[];
}